cannon/mipsevm/tests/difftester.go

```diff
@@ -125,43 +129,72 @@ func (d *DiffTester[T]) run(t testRunner, testCases []T, opts ...TestOption) {
 	cfg := newTestConfig(t, opts...)
 	for _, vm := range cfg.vms {
 		for i, testCase := range testCases {
-			randSeed := randomSeed(t, d.testNamer(testCase), i)
+			randSeed := cfg.randomSeed
+			if randSeed == 0 {
+				randSeed = randomSeed(t, d.testNamer(testCase), i)
+			}
 			mods := d.generateTestModifiers(t, testCase, vm, cfg, randSeed)
 			for _, mod := range mods {
 				testName := fmt.Sprintf("%v%v (%v)", d.testNamer(testCase), mod.name, vm.Name)
 				t.Run(testName, func(t testcaseT) {
 					t.Parallel()
 
-					testDeps := cfg.testDependencies()
-					stateOpts := []mtutil.StateOption{mtutil.WithRandomization(randSeed)}
-					stateOpts = append(stateOpts, d.stateOpts...)
-					goVm := vm.VMFactory(testDeps.po, testDeps.stdOut, testDeps.stdErr, testDeps.logger, stateOpts...)
-					state := mtutil.GetMtState(t, goVm)
-
-					// Set up state
-					r := testutil.NewRandHelper(randSeed * 2)
-					d.initState(t, testCase, state, vm, r)
-					mod.stateMod(state)
+					setup := mod.cachedSetup
+					if setup == nil {
+						setup = d.newTestSetup(t, testCase, vm, cfg, randSeed, mod)
+					}
```

Comment:
To fix performance, we now cache the test setup objects (vm, expected state, etc) that we generate when running `generateTestModifiers()`.